build.py

```diff
@@ -1,40 +1,521 @@
+            "tls": {"termination": "edge"},
+            "to": {"kind": "Service", "name": service, "weight": 100},
+        },
+    }
+    return json.dumps(route)
+
+
+def oc_create_route(project, route, app_url, service):
+    proc = subprocess.Popen(
+        ["oc", "create", "-f", "-"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        stdin=subprocess.PIPE,
+    )
+    r = get_route_def(project, route, app_url, service)
+    print("\n\n")
+    print(r.encode())
+    print("\n\n")
+    proc.communicate(r.encode())
+
+
+def get_pass_configmap(project, cm_name, username, password):
+    configmap = {
+        "apiVersion": "v1",
+        "kind": "ConfigMap",
+        "metadata": {
+            "name": cm_name,
+            "namespace": project,
+            "labels": {"app": project},
+        },
+        "data": {"user-data": username + " " + password},
+    }
+    return json.dumps(configmap)
+
+
+def oc_create_cm_pass(project, cm_name, username, password):
+    proc = subprocess.Popen(
+        ["oc", "create", "-f", "-"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        stdin=subprocess.PIPE,
+    )
+    cm = get_pass_configmap(project, cm_name, username, password)
+    print("\n\n")
+    print(cm.encode())
+    print("\n\n")
+    proc.communicate(cm.encode())
+
+
+def create_objects(
+    openshift_url,
+    openshift_version,
+    app_url,
+    project,
+    service,
+    docker_image,
+    username,
+    password,
+):
+    if not oc_project_exists(project):
+        oc_create_project(project)
+    if not oc_service_account_exists(project, project + "-sa") or not oc_sa_role_exists(
+        project, project + "-sa", "cluster-admin"
+    ):
+        oc_create_service_account(project, project + "-sa", "cluster-admin")
+    if not oc_service_exists(project, service):
+        oc_create_service(project, service)
+    if not oc_route_exists(project, project, app_url):
+        oc_create_route(project, project, app_url, project)
+    cm_name = "admin-pass"
+    # if oc_cm_pass_exists(project, cm_name):
+    #    oc_delete_configmap(project, cm_name)
+    oc_create_cm_pass(project, cm_name, username, password)
+
+    if not oc_dc_exists(project, project):
+        oc_create_dc(openshift_url, openshift_version, project, docker_image, cm_name)
+    else:
+        oc_rollout_dc(project, project)
+
 
 def main():
-    # export MICROSERVER_URL=https://acct-req.k-apps.osh.massopen.cloud 
-    # export MICROSERVER_URL=https://acct-mgt-acct-mgt.s-apps.osh.massopen.cloud
-    microserver_url='https://acct-mgt-acct-mgt.s-apps.osh.massopen.cloud'
-    build_and_deploy()
+    # TODO: make the commandline interface more reasonable
+    #      1) doing a docker build should be optional
+    #      2) generating all of the certificates should be optional
+    if not (len(sys.argv) in [8, 10]):
+        print(len(sys.argv))
+        usage_msg()
+    else:
+        project = str(sys.argv[1])
```

Comment:
The contents of `sys.argv` are already strings. You don't need to explicitly convert them.